examples/object_detection_3d/object_detection_3d/evaluator.py

```diff
@@ -0,0 +1,374 @@
+                offset (float, optional): Offset to set the value range.
+                    Defaults to 0.5.
+                period (float, optional): Period of the value. Defaults to np.pi*2.
+            Returns:
+                (np.ndarray): Value in the range of
+                    [-offset * period, (1-offset) * period]
+            """
+            limited_val = val - np.floor(val / period + offset) * period
+            return limited_val
+
+        dimensions = [bbox.dimensions for bbox in velodyne_bboxes]
+        yaws = limit_period(-np.array([bbox.rotations[2] for bbox in velodyne_bboxes]) - np.pi / 2.0)
+        transformation = np.array(transformation).reshape([4, 4])
+        rectification = np.array(rectification).reshape([4, 4])
+        centers = np.array([list(bbox.center) + [1.0] for bbox in velodyne_bboxes])
+        centers = centers @ (rectification @ transformation).T
+        return [
+            BoundingBox3D(
+                center=tuple(center[:3]),
+                dimensions=(dim[0], dim[2], dim[1]),  # dimension from x, y, z -> x, z, y
+                rotations=(0.0, yaw, 0.0),
+            )
+            for center, dim, yaw in zip(centers, dimensions, yaws)
+        ]
+
+    @staticmethod
+    def ground_truth_to_kitti_format(sample: TestSample, gt: GroundTruth) -> Dict[str, Any]:
+        if len(gt.image_bboxes()) == 0:
+            return dict(
+                name=np.array([]),
+                truncated=np.array([]),
+                occluded=np.array([]),
+                alpha=np.array([]),
+                bbox=np.zeros([0, 4]),
+                dimensions=np.zeros([0, 3]),
+                location=np.zeros([0, 3]),
+                rotation_y=np.array([]),
+            )
+
+        camera_bboxes = KITTI3DEvaluator.transform_to_camera_frame(
+            gt.velodyne_bboxes(),
+            sample.velodyne_to_camera_transformation,
+            sample.camera_rectification,
+        )
+        return dict(
+            name=np.array([bbox.label for bbox in gt.velodyne_bboxes()]),
+            truncated=np.array([bbox.truncated for bbox in gt.velodyne_bboxes()]),
+            occluded=np.array([bbox.occluded for bbox in gt.velodyne_bboxes()]),
+            alpha=np.array(
+                [
+                    KITTI3DEvaluator.alpha_from_bbox3D(bbox, bbox_lidar)
+                    for bbox, bbox_lidar in zip(camera_bboxes, gt.velodyne_bboxes())
+                ],
+            ),
+            bbox=np.array([KITTI3DEvaluator.bbox_to_kitti_format(bbox) for bbox in gt.image_bboxes()]),
+            dimensions=np.array([bbox.dimensions for bbox in camera_bboxes]),
+            location=np.array([KITTI3DEvaluator.center_to_kitti_format(bbox) for bbox in camera_bboxes]),
+            rotation_y=np.array([bbox.rotations[1] for bbox in camera_bboxes]),
+        )
+
+    @staticmethod
+    def inference_to_kitti_format(sample: TestSample, inf: Inference) -> Dict[str, Any]:
```

Comment:
To make the evaluator as simple to understand as possible, I would recommend extracting these static methods to utils in a separate file